testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketHalfClosedTest.java

```diff
@@ -327,8 +328,18 @@ protected void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Excep
                     public void operationComplete(ChannelFuture future) throws Exception {
                         future.channel().close().addListener(new ChannelFutureListener() {
                             @Override
-                            public void operationComplete(ChannelFuture future) throws Exception {
-                                followerCloseLatch.countDown();
+                            public void operationComplete(final ChannelFuture future) throws Exception {
+                                // This is a bit racy but there is no better way how to handle this in Java11.
+                                // The problem is that on close() the underlying FD will not actually be closed directly
+                                // but the close will be done after the Selector did process all events. Because of
```

Comment:
Dumb question: if this test is the only thing using the event loop, this should not be racy right?  If you can limit the selectable things to just this test, there should be nothing more after reaching this point IIUC.

Reply:
The problem is that we depend on the CountdownLatch to tel us at what point we try to do the write that should fail and client / sever use different EventLoops so there is no good way without schedule the countDown() in the future